kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,836 @@
+                            if (direction == 0) {j++ ; }
+                            if (direction == 1) {i++ ; }
+                            if (direction == 2) {j-- ; }
+                            if (direction == 3) {i-- ; }
+                            Node snakeNode(idSnakeNode , knot_vector_u[i], knot_vector_v[j], 0.0);
+                            surrogate_model_part_outer.CreateNewNode(idSnakeNode, snakeNode);
+                            Condition::Pointer pcond1 = surrogate_model_part_outer.CreateNewCondition("LineCondition2D2N", idSnakeNode-1, {{idSnakeNode-1, idSnakeNode}}, p_cond_prop );
+                            idSnakeNode++;
+                            if (direction == 0) {i++ ; }
+                            if (direction == 1) {j-- ; }
+                            if (direction == 2) {i-- ; }
+                            if (direction == 3) {j++ ; }
+                            Node snakeNode2(idSnakeNode , knot_vector_u[i], knot_vector_v[j], 0.0);
+                            surrogate_model_part_outer.CreateNewNode(idSnakeNode, snakeNode2);
+                            Condition::Pointer pcond2 = surrogate_model_part_outer.CreateNewCondition("LineCondition2D2N", idSnakeNode-1, {{idSnakeNode-1, idSnakeNode}}, p_cond_prop );
+                            idSnakeNode++;
+                            if (direction == 0) {j-- ; }
+                            if (direction == 1) {i-- ; }
+                            if (direction == 2) {j++ ; }
+                            if (direction == 3) {i++ ; }
+                            Node snakeNode3(idSnakeNode , knot_vector_u[i], knot_vector_v[j], 0.0);
+                            surrogate_model_part_outer.CreateNewNode(idSnakeNode, snakeNode3);
+                            Condition::Pointer pcond3 = surrogate_model_part_outer.CreateNewCondition("LineCondition2D2N", idSnakeNode-1, {{idSnakeNode-1, idSnakeNode}}, p_cond_prop );
+                            idSnakeNode++;
+                            // Finally rotate the direction of 180 degrees
+                            direction = direction + 2;
+                            if (direction == 4) {direction = 0;}
+                            if (direction == 5) {direction = 1;}
+                        }
+                        else
+                            KRATOS_ERROR << "SnakeSBMUtilities:: " <<  "error in the Snakes Coordinates" << std::endl;
+                    }
+                }
+            }
+            // Check if we have close the snake
+            if (I == start_i+1 && J == start_j+1 && is_special_case == 1 ) {
+                // End of the while loop
+                end = true;
+                }
+        }
+        // Create "fictituos element" to memorize starting and ending node id for each surrogate boundary loop
+        std::vector<ModelPart::IndexType> elem_nodes{1, idSnakeNode-1};
+        IndexType elem_id = surrogate_model_part_outer.GetRootModelPart().Elements().size()+1;
```

Comment:
Watch out: `index != identifier`.

```suggestion
        const ModelPart& r_root = surrogate_model_part_outer.GetRootModelPart();
        IndexType elem_id = r_root.Elements().empty()
                            ? 1
                            : r_root.Elements().back().Id() + 1;
```